File: app/src/main/java/com/lucas/omnia/models/Reply.java
package com.lucas.omnia.models;

import com.google.firebase.database.Exclude;
import com.google.firebase.database.IgnoreExtraProperties;

import java.util.HashMap;
import java.util.Map;

@IgnoreExtraProperties
public class Reply {
    public String uid;
    public String author;
    public long timestamp;
    public String body;
    public boolean edited;
    public int upVoteCount = 0;
    public int downVoteCount = 0;
    public int votesBalance = 0;
    public Map<String, Boolean> upVotes = new HashMap<>();
    public Map<String, Boolean> downVotes = new HashMap<>();

    public Reply() {
        // Default constructor required for calls to DataSnapshot.getValue(Reply.class)
    }

    public Reply(String uid, String author, String body) {
        this.uid = uid;
        this.author = author;
        this.body = body;
    }

    @Exclude
    public Map<String, Object> toMap() {
        HashMap<String, Object> result = new HashMap<>();
        result.put("uid", uid);
        result.put("author", author);
        result.put("timestamp", timestamp);
        result.put("body", body);
        result.put("upVoteCount", upVoteCount);
        result.put("downVoteCount", downVoteCount);
        result.put("votesBalance", votesBalance);

        return result;
    }
}

File: app/src/main/java/com/lucas/omnia/models/Reply.kt
package com.lucas.omnia.models

import com.google.firebase.database.Exclude
import com.google.firebase.database.IgnoreExtraProperties
import java.util.*

@IgnoreExtraProperties
data class Reply(var uid: String? = null, var author: String? = null, var body: String? = null) {
    @JvmField
    var timestamp: Long = 0
    @JvmField
    var edited = false
    @JvmField
    var upVoteCount = 0
    @JvmField
    var downVoteCount = 0
    @JvmField
    var votesBalance = 0
    @JvmField
    var upVotes: MutableMap<String, Boolean> = HashMap()
    @JvmField
    var downVotes: MutableMap<String, Boolean> = HashMap()

    @Exclude
    fun toMap(): Map<String, Any?> {
        val result = HashMap<String, Any?>()
        result["uid"] = uid
        result["author"] = author
        result["timestamp"] = timestamp
        result["body"] = body
        result["upVoteCount"] = upVoteCount
        result["downVoteCount"] = downVoteCount
        result["votesBalance"] = votesBalance
        return result
    }
}